hadoop-hdds/config/src/main/java/org/apache/hadoop/hdds/conf/TimeDurationUtil.java

```diff
@@ -50,6 +50,9 @@ public static long getTimeDurationHelper(String name, String vStr,
     if (null == vUnit) {
       LOG.warn("No unit for " + name + "(" + vStr + ") assuming " + unit);
       vUnit = ParsedTimeDuration.unitFor(unit);
+      if(null == vUnit){
+        throw new NullPointerException("Unexpected unit" + unit);
```

Comment:
I think `IllegalArgumentException` more closely matches the problem than `NullPointerException`.

```suggestion
        throw new IllegalArgumentException("Unexpected unit: " + unit);
```

Reply:
Thanks @adoroszlai for reviewing this too.
After query, `IllegalArgumentException` is indeed more suitable than `NullPointerException`.
I will replace it immediately.